src/runtime_src/core/tools/common/reports/ReportContextHealth.cpp

```diff
@@ -0,0 +1,177 @@
+#include "tools/common/SmiWatchMode.h"
+#include "tools/common/Table2D.h"
+
+// 3rd Party Library - Include Files
+#include <algorithm>
+#include <boost/format.hpp>
+#include <iomanip>
+#include <sstream>
+#include <vector>
+
+using bpt = boost::property_tree::ptree;
+
+void
+ReportContextHealth::
+getPropertyTreeInternal(const xrt_core::device* dev,
+                                            bpt& pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data,
+  // Then update this method to do so.
+  getPropertyTree20202(dev, pt);
+}
+
+void
+ReportContextHealth::
+getPropertyTree20202(const xrt_core::device* dev,
+                                         bpt& pt) const
+{
+  bpt context_health_pt{};
+
+  try {
+    auto context_health_data = xrt_core::device_query<xrt_core::query::context_health_info>(dev);
+
+    // Convert structured data to property tree
+    bpt contexts_array{};
+    for (const auto& context : context_health_data) {
+      bpt context_pt;
+      context_pt.put("txn_op_idx", context.txn_op_idx);
+      context_pt.put("ctx_pc", context.ctx_pc);
+      context_pt.put("fatal_error_type", context.fatal_error_type);
+      context_pt.put("fatal_error_exception_type", context.fatal_error_exception_type);
+      context_pt.put("fatal_error_exception_pc", context.fatal_error_exception_pc);
+      context_pt.put("fatal_error_app_module", context.fatal_error_app_module);
+      contexts_array.push_back(std::make_pair("", context_pt));
+    }
+    context_health_pt.add_child("contexts", contexts_array);
+    context_health_pt.put("context_count", context_health_data.size());
+  } 
+  catch (const std::exception& e) {
+    context_health_pt.put("context_count", 0);
+    context_health_pt.put("error", e.what());
+  }
+
+  // There can only be 1 root node
+  pt.add_child("context_health", context_health_pt);
+}
+
+// Helper function to parse context IDs from elements_filter
+static std::vector<uint32_t> 
+parse_context_ids(const std::vector<std::string>& elements_filter) {
+  std::vector<uint32_t> context_ids;
+  std::stringstream ctx_stream;
+  
+  for (const auto& filter : elements_filter) {
+    if (filter.find("ctx_id=") == 0) {
+      std::string ctx_ids_str = filter.substr(7); // // NOLINT(cppcoreguidelines-avoid-magic-numbers) 
+      ctx_stream.str(ctx_ids_str);
+      ctx_stream.clear(); // Clear error flags
+      std::string ctx_id_token;
+      
+      // Parse comma-separated context IDs
+      while (std::getline(ctx_stream, ctx_id_token, ',')) {
+        try {
+          uint32_t ctx_id = std::stoul(ctx_id_token);
+          context_ids.push_back(ctx_id);
+        } 
+        catch (const std::exception&) {
+          // Invalid ctx_id format, skip this token
+        }
+      }
+      break;
+    }
+  }
+  
+  return context_ids;
+}
+
+static std::string
+generate_context_health_report(const xrt_core::device* dev, const std::vector<std::string>& elements_filter, bool include_timestamp = true) {
```

Comment:
This on the other hand does not fit, so break it.    Not sure why it doesn't align the the suggestion ?
```suggestion
static std::string
generate_context_health_report(const xrt_core::device* dev, 
                                                    const std::vector<std::string>& elements_filter, 
                                                    bool include_timestamp = true) 
{
```